Automattic/sensei
# Allow copying style to all modules, lessons

This is the technically least complex solution to allow the user to use the same style and colors for all blocks. 
Clicking the action, it copies the style customization to all blocks of the same types. 

Does not work on Wordpress 5.3, because half the block editor APIs are missing, and it does not look feasible to work around them. We could approach this as *progressive enhancement*, since it's just an extra quality of life feature.

### Changes proposed in this Pull Request

* Add 'Apply styles to all modules/lessons' button to the Advanced settings panel, copying the styles of the selected block
* Copies colors and block style for modules
* Copies colors and font size for lessons

### Testing instructions

* Have a course outline with multiple modules and lessons
* Change block style and colors of a module, click `Apply style to all modules`
* Module customization should be synced
* Change colors and typography of a lesson, click `Apply style to all lessons`
* Lesson customization should be synced


### Screenshot / Video

(Video is a bit older with the setting being under Color settings)

![QuickTime movie](https://user-images.githubusercontent.com/176949/94961481-0312e280-04f5-11eb-8a63-045c5fbcd717.gif)
<img width="278" alt="image" src="https://user-images.githubusercontent.com/176949/94964923-d5c93300-04fa-11eb-939c-ae51dddbbdc2.png">
